### ISSUE TITLE
Slicing .uns dictionaries creates reversed order of indices with scipy < 1.3.0

### ISSUE THREAD

**Issue by LuckyMD** [theislab/anndata#165]
Hey!

@flying-sheep and I just found some strange behaviour when subsetting an AnnData object. The sparse dictionaries stored in `adata.uns` have an inverted ordering of indices after slicing, while the ordering of indices in the original object is ascending as expected. A quick reproducible example would be:

```
import scanpy as sc
import numpy as np

adata = sc.datasets.blobs()
sc.pp.pca(adata)
sc.pp.neighbors(adata)
sc.tl.louvain(adata)

adata_sub = adata[np.isin(adata.obs['louvain'], ['0', '1']),:].copy()

print(adata.uns['neighbors']['connectivities'])
print(adata_sub.uns['neighbors']['connectivities'])
```

The resulting ordering of columns is inverted. For example:
```
print(adata.uns['neighbors']['connectivities'])
  (0, 41)	0.7349735270358434
  (0, 46)	0.23910809145417686
  (0, 118)	0.4013338135392787
  (0, 134)	0.5041749591507215
  (0, 143)	1.0
  (0, 169)	0.07222835175020617
...
 ```

vs
```
print(adata_sub.uns['neighbors']['connectivities'])
  (0, 255)	0.25173951052835253
  (0, 245)	0.044398562560975484
  (0, 238)	0.6661832487525872
  (0, 234)	0.11444020003955598
  (0, 214)	0.049137678749762444
  (0, 199)	1.0
  (0, 168)	0.6406455197199623
  (0, 162)	0.17288532762283385
  (0, 149)	0.23061565871146353
...
```
This doesn't occur for all environments. I have this error for: 
scanpy==1.4.3 anndata==0.6.21 umap==0.3.9 numpy==1.16.4 scipy==1.2.1 pandas==0.24.1 scikit-learn==0.20.2 statsmodels==0.9.0 python-igraph==0.7.1 louvain==0.6.1 

But not for:
scanpy==1.4.3+18.g7266e67 anndata==0.6.20 umap==0.3.8 numpy==1.16.3 scipy==1.3.0 pandas==0.24.2 scikit-learn==0.20.0 statsmodels==0.9.0 python-igraph==0.7.1 louvain==0.6.1 

This could have something to do with the new scipy, which we're currently telling people to avoid (and I can't use it on the server). I updated scanpy, anndata, and numpy to test if it had to do with old versions of these packages, but that doesn't appear to be the case. My laptop environment works fine, the server one doesn't (and I can't test scipy 1.3.0 on there).

Oh... and for context: This is an issue as `anndata2ri` fails under these circumstances.

**Comment by LuckyMD** [comment]
Just confirmed this is a scipy 1.2.1 issue. Downgraded my local env to scipy 1.2.1 and got the issue.

**Comment by flying-sheep** [comment]
The IndexMixin issue is long fixed, statsmodels/statsmodels#5759 is the problem.

**Comment by LuckyMD** [comment]
Great! Will give this a go tomorrow!
